--- check_multi-0.26/plugins/check_multi.in~	2012-06-25 21:58:06.000000000 +0300
+++ check_multi-0.26/plugins/check_multi.in	2012-06-25 22:41:23.306027423 +0300
@@ -477,7 +477,7 @@
 		return $UNKNOWN;
 	}
 	#--- 2. check if tmp_dir has correct permissions
-	if  ((stat("$opt{set}{tmp_dir}"))[2] != oct("$opt{set}{tmp_dir_permissions}") &&
+	if  (((stat("$opt{set}{tmp_dir}"))[2] & 07777) != oct("$opt{set}{tmp_dir_permissions}") &&
 		! chmod(oct("$opt{set}{tmp_dir_permissions}"), "$opt{set}{tmp_dir}") ) {
 		print "Error: could not set tmp directory $opt{set}{tmp_dir} permissions to $opt{set}{tmp_dir_permissions} as user $opt{set}{user}\n";
 		return $UNKNOWN;
--- check_multi-0.26/configure.ac~	2011-11-18 21:28:13.000000000 +0200
+++ check_multi-0.26/configure.ac	2012-06-25 22:44:59.861420799 +0300
@@ -83,7 +83,7 @@
 AC_ARG_WITH(extinfo_in_status,		[  --with-extinfo_in_status=<0|1>           display check_multi output in status.cgi],extinfo_in_status=$withval,extinfo_in_status=0)
 AC_ARG_WITH(feed_passive_autocreate,	[  --with-feed_passive_autocreate=<0|1>>    flag, if feed passive service definitions should be created automatically (default:1)],feed_passive_autocreate=$withval,feed_passive_autocreate="1")
 AC_ARG_WITH(feed_passive_dir,		[  --with-feed_passive_dir=<path>           directory for automatically created feed passive service definitions (default:${sysconfdir}/check_multi/feed_passive)],feed_passive_dir=$withval,feed_passive_dir="${sysconfdir}/check_multi/feed_passive")
-AC_ARG_WITH(feed_passive_dir_permissions,[  --with-feed_passive_dir_permissions=<octal perms> permissions for feed passive config dir (default:41777)],feed_passive_dir_permissions=$withval,feed_passive_dir_permissions="41777")
+AC_ARG_WITH(feed_passive_dir_permissions,[  --with-feed_passive_dir_permissions=<octal perms> permissions for feed passive config dir (default:41777)],feed_passive_dir_permissions=$withval,feed_passive_dir_permissions="01777")
 AC_ARG_WITH(findbin,			[  --with-findbin=<0|1>                     use Perl module FindBin for addressing plugins (default:1)],findbin=$withval,findbin=1)
 AC_ARG_WITH(file_extension,		[  --with-file_extension=<ext>              default extension for check_multi command files (default:"cmd")],file_extension=$withval,file_extension="cmd")
 AC_ARG_WITH(hostname,			[  --with-hostname=<host>                   hostname of nagios server for mail / web purposes (default:"localhost")],hostname=$withval,hostname="localhost")
@@ -116,7 +116,7 @@
 AC_ARG_WITH(tag_notes_link,		[  --with-tag_notes_link=<URL>              documentation URL for child checks (default:"")],tag_notes_link=$withval,tag_notes_link="")
 AC_ARG_WITH(target,			[  --with-target=<target_frame>             HTML target frame for URLs (default:_self)],target=$withval,target="_self")
 AC_ARG_WITH(tmp_dir,			[  --with-tmp_dir=<path>                    /path/to/tmp_dir (default:/tmp/check_multi)],tmp_dir=$withval,tmp_dir="/tmp/check_multi")
-AC_ARG_WITH(tmp_dir_permissions,	[  --with-tmp_dir_permissions=<octal perms> permissions of tmp_dir (default:41777)],tmp_dir_permissions=$withval,tmp_dir_permissions="41777")
+AC_ARG_WITH(tmp_dir_permissions,	[  --with-tmp_dir_permissions=<octal perms> permissions of tmp_dir (default:41777)],tmp_dir_permissions=$withval,tmp_dir_permissions="01777")
 AC_ARG_WITH(tmp_etc,			[  --with-tmp_etc=<path>                    /path/to/tmp_dir/etc (default:$tmp_dir/etc)],tmp_etc=$withval,tmp_etc="$tmp_dir/etc")
 AC_ARG_WITH(parent_timeout,		[ 
 major settings:
