--- check_multi-0.26/sample_config/nagiostats/nagiostats.cmd.in~	2011-11-18 21:28:13.000000000 +0200
+++ check_multi-0.26/sample_config/nagiostats/nagiostats.cmd.in	2012-06-25 23:54:25.468638432 +0300
@@ -8,6 +8,8 @@
 # Call: check_multi -f nagiostats.cmd -s NSTATS=/path/to/nagiostats
 #
 
+eval [ NSTATS ] = ("$NSTATS$" eq "") ? "@sbindir@/nagiostats" : "$NSTATS$"
+
 # 1. check correct parameter passing for NSTAT binary
 eval    [ NSTATS_defined                ] =
 	if (! "$NSTATS$") {
